code/modules/projectiles/projectile/trace.dm

```diff
@@ -0,0 +1,36 @@
+//Helper proc to check if you can hit them or not.
+/proc/check_trajectory(atom/target as mob|obj, atom/firer as mob|obj, var/pass_flags=PASS_FLAG_TABLE|PASS_FLAG_GLASS|PASS_FLAG_GRILLE)
+	if(!istype(target) || !istype(firer))
+		return 0
+
+	var/obj/item/projectile/test/trace = new /obj/item/projectile/test(get_turf(firer)) //Making the test....
+
+	//Set the flags and pass flags to that of the real projectile...
+	trace.pass_flags = pass_flags
+
+	return trace.launch(target) //Test it!
+
+/obj/item/projectile/proc/_check_fire(atom/target as mob, var/mob/living/user as mob)  //Checks if you can hit them or not.
```

Comment:
What's this for? I don't see any references to it.

Reply:
I believe it's used for raycasting projectiles, I think?

Reply:
Correction, this does nothing. Removing.